### ISSUE TITLE
linter: False positive of no-unused-private-class-members when used in default paramater

### ISSUE THREAD

**Issue by oriSomething** [oxc-project/oxc#14775]
### What version of Oxlint are you using?

1.23.0

### What command did you run?

yarn exec oxlint --config .oxlintrc.json --type-aware

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["typescript"],
  "categories": {
    "correctness": "off",
    "pedantic": "off",
    "nursery": "off",
    "perf": "off",
    "restriction": "off",
    "style": "off",
    "suspicious": "off"
  },
  "rules": {
    "no-unused-private-class-members": "error"
  }
}

```

### What happened?

For the code:
```ts

export class A {
  #x;

  constructor(x: number) {
    this.#x = x;
  } 

  get(y = this.#x) {
    return y;
  }
}
```

I get the error:
```
'x' is defined but never used.oxc[eslint(no-unused-private-class-members)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-private-class-members.html)
```

**Comment by camc314** [comment]
Thank for reporing, will be fixed by #14801
